backup-fs-vms.sh

```diff
@@ -118,6 +123,58 @@ echo "rabe-backup CTRL-C catched"
 exit 1
 }
 
+# function to handle rsync return codes
+handle_rsync_ret() {
+  declare ret="$1" vm="$2" src="$3" dst="$4"
+
+  if [[ "${ret}" -eq "0" ]]; then
+    echo "rabe-backup Sync of ${src} to ${dst} successful!"
+  elif [[ "${ret}" -eq "23" ]]; then
+    echo "rabe-backup INFO: ${src} does not exist."
+  elif [[ "${ret}" -eq "12" ]]; then
+    echo "rabe-backup ERROR: Permission denied on ${src}."
+    (("errors_vm++"));
+  elif [[ "${ret}" -eq "255" ]]; then
+    echo "rabe-backup ERROR: Host ${vm}.vm-admin.int.rabe.ch is not online or could not be resolved."
+    (("errors_vm++"));
+  else
+    echo "rabe-backup ERROR: Unknown error (${ret}) occured when trying to rsync $src."
+    (("errors_vm++"));
+  fi
+}
+
+# function to remotely fetch content of a file
+cat_remote_file() {
+  declare vm="$1" file="$2"
+
+  ${RSH_CMD} "${vm}.vm-admin.int.rabe.ch" "cat ${file}"
+}
+
+# function to backup custom directories specified by the remote vm
+backup_custom_dirs() {
+  declare vm="$1"
+
+  # fetch include and exclude files
+  local includes=$(cat_remote_file "${vm}" "${REMOTE_INCLUDE}")
+  local excludes=$(cat_remote_file "${vm}" "${REMOTE_EXCLUDE}")
+
+  # `--relative` is required to exclude absolute paths
+  # https://superuser.com/a/625231
+  local exclude_opts="--relative"
+  for exclude in $excludes; do
+    exclude_opts="${exclude_opts} --exclude $exclude"
```

Comment:
Is the --relative option not duplicated here?
Is exclude_opts really extended here?
I'm not sure if this works here.

Reply:
I added `exclude_opts="${exclude_opts} --exclude ${exclude}"` in 9e8b3321fbd5b7a9733a2b5b725bfb55dd4401f0.